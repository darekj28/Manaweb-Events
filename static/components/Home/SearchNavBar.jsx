var React = require('react');
var Link = require('react-router').Link;
import NotificationsDropdown from "../GenericNavBar/NotificationsDropdown.jsx";
import AccountDropdown from "../GenericNavBar/AccountDropdown.jsx";
import FilterButton from "./FilterButton.jsx";

function contains(collection, item) {
	if(collection.indexOf(item) !== -1) return true;
	else return false;
}

export default class SearchNavBar extends React.Component {
	componentDidMount() {
		var searchVisible = 0;
		$('.navbar-search-form').show();		

		// $('.navbar-search-form').hide();
		// $('[data-toggle="search"]').click(function(){
	 //        if(searchVisible == 0){
	 //            searchVisible = 1;
	 //            $(this).parent().addClass('active');
	 //            $('.navbar-search-form').fadeIn(function(){
	 //                $('.navbar-search-form input').focus();
	 //            });
	 //        } else {
	 //            searchVisible = 0;
	 //            $(this).parent().removeClass('active');
	 //            $(this).blur();
	 //            $('.navbar-search-form').fadeOut(function(){
	 //                $('.navbar-search-form input').blur();
	 //            });
	 //        } 
	 //    });
	    $('#searchInput').keypress(function(event) {
	    	if(event.keyCode == 13){ 
			   event.preventDefault();
			}
	    });
	    $('.SearchNavBarGlyphicon').focus(function() {
	    	$(this).blur();
	    });
	}
	handleResetFilterUser() {
		this.props.handleFilterUser(this.props.userIdToFilterPosts);
	}
	handleSearch() {
		this.props.onSearch(this.searchText.value);
	}
	render() {
		return (
			<nav className="navbar navbar-default" role="navigation">
				<div className="container">
					<div className="navbar-header">
						<button type="button" className="SearchNavBarGlyphicon navbar-toggle" 
										data-toggle="collapse" 
										data-target="#bs-example-navbar-collapse-1">
				            <span className="sr-only">Toggle navigation</span>
				            <span className="icon-bar"></span>
				            <span className="icon-bar"></span>
				            <span className="icon-bar"></span>
				          </button>
				          <Link to="/" onClick={this.handleResetFilterUser.bind(this)} 
				          				className="SearchNavBarGlyphicon 
				          								navbar-brand navbar-brand-logo">
				                <span className="glyphicon glyphicon-home"></span>
				              </Link>
				        </div>
				        <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
				          <ul className="nav navbar-nav navbar-right">
		                   {/*  <li>
		                        <a href="javascript:void(0)" data-toggle="search">
		 							<span className="glyphicon glyphicon-search"></span>
		                        </a>
		                    </li>
		                */}
				          	<NotificationsDropdown/>
				          	<AccountDropdown currentUser = {this.props.currentUser} 
				          				name={this.props.name}/>
				          </ul>
				         <form className="navbar-form navbar-right navbar-search-form" 
				         		role="search">                  
			                 <div className="form-group">
			                 	  <div className="input-group input-group-unstyled">
				                      <input type="text" value={this.props.searchText} 
				                      			ref={(input) => this.searchText = input} 
				                      			id="searchInput" className="form-control" 
				                      			placeholder="Search..." 
				                      			onChange={this.handleSearch.bind(this)}/>
				                      <div className = "input-group-addon"></div>
								  	  {this.props.actions.map(function(action, i) {
											var button = !this.props.isComment ? 
											<FilterButton key={i} 
													onClick={this.props.handleFilterClick} 
													selected={true} isSearch={true} name={action}/> : '';
											return button;
										}, this)}
								  </div>
								  
			                 </div> 
			              </form>
				        </div>
				      </div>
				    </nav>
		)
	}
}