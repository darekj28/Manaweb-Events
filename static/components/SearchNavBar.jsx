var React = require('react');
var Link = require('react-router').Link;
// var $ = require('jquery');
import NotificationsDropdown from "./NotificationsDropdown.jsx";
import AccountDropdown from "./AccountDropdown.jsx";
import FilterButton from "./FilterButton.jsx";

function contains(collection, item) {
	if(collection.indexOf(item) !== -1) return true;
	else return false;
}

export default class SearchNavBar extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			notifications : [],
			numUnseen : ''
		};
		this.getNotifications = this.getNotifications.bind(this);
        this.seeNotifications = this.seeNotifications.bind(this);
		this.handleSearch = this.handleSearch.bind(this);
		this.handleResetFilterUser = this.handleResetFilterUser.bind(this);
		this.handleResetFilterButtons = this.handleResetFilterButtons.bind(this);
	}
	componentDidMount() {
		var searchVisible = 0;
		$('.navbar-search-form').show();		

		// $('.navbar-search-form').hide();
		// $('[data-toggle="search"]').click(function(){
	 //        if(searchVisible == 0){
	 //            searchVisible = 1;
	 //            $(this).parent().addClass('active');
	 //            $('.navbar-search-form').fadeIn(function(){
	 //                $('.navbar-search-form input').focus();
	 //            });
	 //        } else {
	 //            searchVisible = 0;
	 //            $(this).parent().removeClass('active');
	 //            $(this).blur();
	 //            $('.navbar-search-form').fadeOut(function(){
	 //                $('.navbar-search-form input').blur();
	 //            });
	 //        } 
	 //    });
	    $('#searchInput').keypress(function(event) {
	    	if(event.keyCode == 13){ 
			   event.preventDefault();
			}
	    });
	    $('.SearchNavBarGlyphicon').focus(function() {
	    	$(this).blur();
	    });
	}
	getNotifications() {
		$.post('/getNotifications', 
            function(data) {
                var notifications = [];
                var count = 0;
                data.notification_list.map(function(obj) {
                    if (!obj['seen']) count++; 
                    notifications.unshift({
                        comment_id : obj['comment_id'],
                        notification_id : obj['notification_id'],
                        timeString : obj['timeString'],
                        sender_id : obj['sender_id'],
                        action : obj['action'],
                        receiver_id : obj['receiver_id'],
                        seen : obj['seen']
                    });
                });
                this.setState({notifications : notifications});
                this.setState({numUnseen: String(count)});
            }.bind(this));
		this.seeNotifications();
	}
	seeNotifications() {
        this.state.notifications.map(function (obj){
            $.post('/seeNotification', {notification_id: obj['notification_id']})
        });
    }
	handleResetFilterUser() {
		this.props.handleFilterUser(this.props.userIdToFilterPosts);
	}
	handleResetFilterButtons() {
		this.props.actions.map(function(action) {
			if (contains(filters, action)) this.props.handleFilterClick(action, true);
		});
	}
	handleSearch() {
		this.props.onSearch(this.searchText.value);
	}
	render() {
		var that = this;
		return (
			<nav className="navbar navbar-default" role="navigation">
				<div className="container">
					<div className="navbar-header">
						<button type="button" className="SearchNavBarGlyphicon navbar-toggle" data-toggle="collapse" 
										data-target="#bs-example-navbar-collapse-1">
				            <span className="sr-only">Toggle navigation</span>
				            <span className="icon-bar"></span>
				            <span className="icon-bar"></span>
				            <span className="icon-bar"></span>
				          </button>
				          <Link to="/" onClick={this.handleResetFilterUser} className="SearchNavBarGlyphicon navbar-brand navbar-brand-logo">
				                <span className="glyphicon glyphicon-home"></span>
				              </Link>
				        </div>
				        <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
				          <ul className="nav navbar-nav navbar-right">
		                   {/*  <li>
		                        <a href="javascript:void(0)" data-toggle="search">
		 							<span className="glyphicon glyphicon-search"></span>
		                        </a>
		                    </li>
		                */}
				          	<NotificationsDropdown notifications={this.state.notifications} numUnseen={this.state.numUnseen} getNotifications={this.getNotifications}/>
				          	<AccountDropdown name={this.props.name}/>
				          </ul>
				         <form className="navbar-form navbar-right navbar-search-form" role="search">                  
			                 <div className="form-group">
			                 	  <div className="input-group input-group-unstyled">
				                      <input type="text" value={this.props.searchText} ref={(input) => this.searchText = input} 
				                      			id="searchInput" className="form-control" placeholder="Search..." 
				                      			onChange={this.handleSearch}/>
				                      <div className = "input-group-addon"></div>
								  	  {this.props.actions.map(function(action, i) {
											var button = !that.props.isComment ? <FilterButton key={i} onClick={that.props.onClick} active={true} isSearch={true} name={action}/> : '';
											return button;
										})}
								  </div>
								  
			                 </div> 
			              </form>
				        </div>
				      </div>
				    </nav>
		)
	}
}