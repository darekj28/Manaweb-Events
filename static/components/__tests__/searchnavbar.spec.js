import SearchNavBar from '../SearchNavBar.jsx';

describe("SearchNavBar", () => {
	it("renders correctly", () => {
		const name = "Jay Kim";
		const actions = ["Trade", "Play", "Chill"];
		const snb = renderer.create(
			<SearchNavBar searchText="a" actions={actions} name={name}
				onSearch={jest.fn()} onClick={jest.fn()}/>
			);
		expect(snb).toMatchSnapshot();
	});
	it("should call onSearch with text in textbox when user types", () => {
		expect(true).toBe(true);
	});
});