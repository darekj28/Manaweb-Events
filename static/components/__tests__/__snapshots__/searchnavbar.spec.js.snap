exports[`SearchNavBar renders correctly 1`] = `
<nav
  className="navbar navbar-default"
  role="navigation">
  <div
    className="container">
    <div
      className="navbar-header">
      <button
        className="navbar-toggle"
        data-target="#bs-example-navbar-collapse-1"
        data-toggle="collapse"
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
      <a
        className="navbar-brand navbar-brand-logo"
        onClick={[Function]}
        style={Object {}}>
        <span
          className="glyphicon glyphicon-home" />
      </a>
    </div>
    <div
      className="collapse navbar-collapse"
      id="bs-example-navbar-collapse-1">
      <ul
        className="nav navbar-nav navbar-right">
        <li>
          <a
            data-toggle="search"
            href="javascript:void(0)">
            <span
              className="glyphicon glyphicon-search" />
          </a>
        </li>
        <li
          className="dropdown">
          <a
            className="dropdown-toggle"
            data-toggle="dropdown"
            href="#">
            <span
              className="glyphicon glyphicon-envelope" />
          </a>
          <ul
            className="dropdown-menu">
            <li>
              <a
                href="#">
                Insert notifications here.
              </a>
            </li>
            <li>
              <a
                href="#">
                Insert notifications here.
              </a>
            </li>
            <li>
              <a
                href="#">
                Insert notifications here.
              </a>
            </li>
          </ul>
        </li>
        <li
          className="dropdown">
          <a
            className="dropdown-toggle"
            data-toggle="dropdown"
            href="#">
            <span
              className="glyphicon glyphicon-user" />
          </a>
          <ul
            className="dropdown-menu">
            <li>
              <a
                href="#">
                Jay Kim
              </a>
            </li>
            <li
              className="divider" />
            <li>
              <a
                href="/settings">
                Settings
              </a>
            </li>
            <li>
              <a
                href="#">
                Your Posts
              </a>
            </li>
            <li>
              <a
                href="/logout">
                Logout
              </a>
            </li>
          </ul>
        </li>
      </ul>
      <form
        className="navbar-form navbar-right navbar-search-form"
        role="search">
        <div
          className="form-group">
          <div
            className="input-group input-group-unstyled">
            <input
              className="form-control"
              id="searchInput"
              onChange={[Function]}
              placeholder="Search..."
              type="text"
              value="a" />
            <div
              className="input-group-addon" />
            <a
              className="input-group-addon">
              <span
                className="glyphicon glyphicon-transfer icon-success filterButton"
                onClick={[Function]} />
            </a>
            <a
              className="input-group-addon">
              <span
                className="glyphicon glyphicon-play icon-success filterButton"
                onClick={[Function]} />
            </a>
            <a
              className="input-group-addon">
              <span
                className="glyphicon glyphicon-time icon-success filterButton"
                onClick={[Function]} />
            </a>
          </div>
        </div>
      </form>
    </div>
  </div>
</nav>
`;
